### PR TITLE
[inductor] Write generated files from parent process

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #123012
* #123011
* #122941
* __->__ #123409

Before this PR we would pass generated source code over a pipe to the compile worker then the compile worker would write out the file.  Doing it this way is faster and results in smaller messages to the workers (and lets us skip creating the workers in the warm start case).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -2804,18 +2879,18 @@ def wait(self, scope: Dict[str, Any]) -> None:
...
                     pbar.update(1)
 
         _compile_end()
 
 
-if os.environ.get("TORCH_TNT_IN_USE", "0") == "1":
-    # When TorchTNT is used, calling warm_pool() here will cause the
-    # compile workers created not being able to be shut down inside
-    # shutdown_compile_workers(). This may cause significant QPS drop.
-    log.info("Do not call AsyncCompile.warm_pool() because TorchTNT is in use.")
```
Comment (desertfire) [inline torch/_inductor/codecache.py:2818]: Should we keep this `log.info`, or is it fixed by this PR?

**Review by desertfire — APPROVED** [review]
I would recommend to ghimport, just to make sure it works internally as well.

How much compilation time speedup does this PR bring?

File: torch/_inductor/codecache.py (hunk 1 above)
Comment (jansel) [reply torch/_inductor/codecache.py:2818]: Doesn't seem that useful to me

**Comment by DanilBaibak** [comment]
Hi @jansel! Sorry to revert your changes, but the lower PR in the stack #122661 broke internal build jobs.

**Pull request closed without merging**